machines-endpoints/pkg/machines-endpoints/main.go

```diff
@@ -346,6 +351,69 @@ func localHTTPClient() *http.Client {
 	}
 }
 
+func createMachinesList(machines []Machine) (string, error) {
+	type Machine struct {
+		Serial string `json:"serial"`
+		BmcIP  string `json:"bmc_ip"`
+		NodeIP string `json:"ipv4"`
+		Role   string `json:"role"`
+		State  string `json:"state"`
+	}
+	var ml []Machine
+
+	for _, machine := range machines {
+		var m Machine
+		if machine.Spec.BMC.IPv4 == "" {
+			continue
+		}
+		m.Serial = machine.Spec.Serial
+		m.BmcIP = machine.Spec.BMC.IPv4
+		m.NodeIP = machine.Spec.IPv4[0]
+		m.Role = machine.Spec.Role
+		m.State = machine.Status.State
+		ml = append(ml, m)
+	}
+
+	byteJSON, err := json.Marshal(ml)
+	if err != nil {
+		return "", err
+	}
+	return string(byteJSON), nil
+}
+
+func (c client) updateBmcLogCollectorConfigMap(ctx context.Context, machinesList string) error {
+	data := make(map[string]string)
+	data["machineslist.json"] = machinesList
+
+	ns, _, err := c.kubeConfig.Namespace()
+	if err != nil {
+		return err
+	}
+
+	nextConfigMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: bmcLogCollectorConfigMapName,
+		},
+		Data: data,
+	}
+
+	cms := c.k8s.CoreV1().ConfigMaps(ns)
+	currentConfigMap, err := cms.Get(ctx, bmcLogCollectorConfigMapName, metav1.GetOptions{})
+	if reflect.DeepEqual(nextConfigMap.Data, currentConfigMap.Data) {
+		return nil
+	}
```

Comment:
Please check errors before using `currentConfigMap`.

Reply:
Changed this
```
if reflect.DeepEqual(nextConfigMap.Data, currentConfigMap.Data) || err == nil {
```

Reply:
changed